bingo.js

```diff
@@ -453,7 +485,8 @@ function toggleStreamerMode()
 	$(".dropdown").hide();
 	hideOptionsMenu();
 	updateStreamerMode();
-	pushNewUrl();
+	pushNewUrl(generateNewUrlParamWithProgress());
+	updateCurrentSheetProgress();
```

Comment:
It is suspicious that `toggleStreamerMode` and `changeVersion` below have a call to `updateCurrentSheetProgress`, but a very similar function `toggleHidden` did not get the same treatment.

Reply:
And a similar question about function `newSeed`. Whenever version, difficulty, or seed change (components of local storage key – see function `getKeyForCurrentProgress`) the progress should be loaded from local storage.

If a user opens a URL which has saved progress, it means that they are still playing the sheet. If they are still playing the sheet, it means that the save should be preserved. To preserve the save should be protected from being deleted. We delete the oldest saves, therefore whenever the progress is loaded from local storage, we should update its timestamp in local storage, i.e. re-save it with `Date.now()`.

Reply:
Another bug to demonstrate the issue. Steps:

1. Click on some squares.
2. Change difficulty
3. Change the difficulty back
4. Toggle streamer mode
5. Observe local storage in developer tools

Because `changeDifficulty` doesn't load the progress, the sheet is blank after step 3. Because `toggleStreamerMode` rewrites the progress with current sheet, after step 4, the progress of step 1 is overwritten with a blank sheet.

Reply:
> Whenever version, difficulty, or seed change ... the progress should be loaded from local storage.

Changes to version, difficulty, or seed cause `generateNewSheet();` to be called. So it seems that instead of adding loading of progress to the function `loadSettings()`, it should be added to the very end of the function `generateNewSheet()` instead.

Reply:
Now that I thought this through, back to the original suspicion:

> It is suspicious that `toggleStreamerMode` and `changeVersion` below have a call to `updateCurrentSheetProgress`, but a very similar function `toggleHidden` did not get the same treatment.

None of these need a call to `updateCurrentSheetProgress`. Streamer mode and hidden/shown bingo don't affect progress. On the other hand, version and the other two parameters (difficulty and seed) should cause the progress to be loaded (see the previous comment).

So, in total there should be two calls to `updateCurrentSheetProgress();`:

 1. in `setSquareColor` (already in the code)
 2. in `processSheetAndStoreAs` to [update the timestamp](https://github.com/Joshimuz/mcbingo/pull/121#discussion_r611969291).